dvc/dependency/repo.py

```diff
@@ -1,43 +1,74 @@
 from __future__ import unicode_literals
 
-import os
 import copy
 
-from dvc.utils.compat import urlparse
-from dvc.external_repo import ExternalRepo
+from funcy import merge
+from schema import Optional
+from contextlib import contextmanager
+
+from dvc.external_repo import external_repo
 
 from .local import DependencyLOCAL
 
 
 class DependencyREPO(DependencyLOCAL):
     PARAM_REPO = "repo"
+    PARAM_URL = "url"
+    PARAM_REV = "rev"
+    PARAM_REV_LOCK = "rev_lock"
 
-    def __init__(self, erepo, stage, *args, **kwargs):
-        self.erepo = ExternalRepo(stage.repo.dvc_dir, **erepo)
-        super(DependencyLOCAL, self).__init__(stage, *args, **kwargs)
-
-    def _parse_path(self, remote, path):
-        self.erepo.install(self.repo.cache.local.cache_dir)
+    REPO_SCHEMA = {
+        Optional(PARAM_URL): str,
+        Optional(PARAM_REV): str,
+        Optional(PARAM_REV_LOCK): str,
+    }
 
-        out_path = os.path.join(
-            self.erepo.repo.root_dir, urlparse(path).path.lstrip("/")
-        )
+    def __init__(self, def_repo, stage, *args, **kwargs):
+        self.def_repo = def_repo
+        super(DependencyREPO, self).__init__(stage, *args, **kwargs)
 
-        out, = self.erepo.repo.find_outs_by_path(out_path)
-        self.info = copy.copy(out.info)
-        self._erepo_stage = copy.copy(out.stage.path)
-        return self.REMOTE.path_cls(out.cache_path)
+    def _parse_path(self, remote, path):
+        return None
 
     @property
     def is_in_repo(self):
         return False
 
+    def __str__(self):
+        return "{} ({})".format(self.def_path, self.def_repo[self.PARAM_URL])
+
+    @contextmanager
+    def make_repo(self, **overrides):
+        with external_repo(**merge(self.def_repo, overrides)) as repo:
+            yield repo
+
+    def status(self):
```

Comment:
are we running status for the locked stage?

Reply:
We do, but not for the dependencies. So `update available` will only be visible if you `dvc unlock import.dvc`

Reply:
is there a way to check the status before updating it then?

Reply:
Yes, if you `dvc unlock import.dvc` first and then run `dvc status`

Reply:
Hmm ... not an easy interface ... don't see an easy way to make it better ... may be make `dvc status target.dvc` ignore the lock?

Reply:
@shcheklein Maybe `dvc update something.dvc --dry` then? Changing `dvc status` itself doesn't feel right.

Reply:
@shcheklein In any case, that additional command can be introduced on top, as it doesn't affect the rest of the logic.

Reply:
`--dry` sounds like a good start

Reply:
Why status ignores locked stages? This doesn't make sense from consistency POV only from optimization one. Is locked about optimization?

Reply:
Yes, `locked` is also about not actively going and checking dependencies of the stage(e.g. not re-downloading external repo, not going to s3 to check for external s3 dependency etc). Also, `locked` is kinda making a data source from any stage, so showing status for its dependencies doesn't seem like a desired behaviour.